storm-server/src/main/java/org/apache/storm/DaemonConfig.java

```diff
@@ -1023,6 +1023,43 @@
     public static String STORM_SUPERVISOR_MEDIUM_MEMORY_GRACE_PERIOD_MS =
         "storm.supervisor.medium.memory.grace.period.ms";
 
+    /**
+     * Class implementing MetricStore.
+     */
+    @NotNull
+    @isString
+    public static final String STORM_METRIC_STORE_CLASS = "storm.metricstore.class";
+
+    /**
+     * RocksDB file location.
+     */
+    @isString
+    public static final String STORM_ROCKSDB_LOCATION = "storm.metricstore.rocksdb.location";
+
+    /**
+     * RocksDB create if missing flag.
+     */
+    @isBoolean
+    public static final String STORM_ROCKSDB_CREATE_IF_MISSING = "storm.metricstore.rocksdb.create_if_missing";
+
+    /**
+     * RocksDB metadata cache capacity.
+     */
+    @isInteger
+    public static final String STORM_ROCKSDB_METADATA_STRING_CACHE_CAPACITY = "storm.metricstore.rocksdb.metadata_string_cache_capacity";
+
+    /**
+     * RocksDB setting for length of metric retention.
+     */
+    @isInteger
+    public static final String STORM_ROCKSDB_METRIC_RETENTION_HOURS = "storm.metricstore.rocksdb.retention_hours";
+
+    /**
+     * RocksDB setting for period of metric deletion thread.
+     */
+    @isInteger
+    public static final String STORM_ROCKSDB_METRIC_RETENTION_PERIOD_HOURS = "storm.metricstore.rocksdb.retention_period_hours";
```

Comment:
Could we add in deletion to the name of the config somewhere?  I am not sure that it is that clear what the config does from just the name of it.